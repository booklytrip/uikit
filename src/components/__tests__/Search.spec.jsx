import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from '../Search';

describe('Search', () => {
    it('renders without crashing', () => {
        mount(<Search />);
    });
    
    describe('Snapshots', () => {
        it('renders with placeholder', () => {
            const rendered = renderer.create(<Search placeholder="Placeholder" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders in loading state', () => {
            const rendered = renderer.create(<Search loading />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different size', () => {
            const rendered = renderer.create(<Search size="small" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
