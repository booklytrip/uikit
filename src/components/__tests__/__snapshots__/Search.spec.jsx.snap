exports[`Search Snapshots renders in loading state 1`] = `
<div
  className="ui search loading">
  <div
    className="ui icon input">
    <input
      className="prompt"
      placeholder={undefined}
      type="text" />
    <i
      className="search icon" />
  </div>
</div>
`;

exports[`Search Snapshots renders with different size 1`] = `
<div
  className="ui search small">
  <div
    className="ui icon input">
    <input
      className="prompt"
      placeholder={undefined}
      type="text" />
    <i
      className="search icon" />
  </div>
</div>
`;

exports[`Search Snapshots renders with placeholder 1`] = `
<div
  className="ui search">
  <div
    className="ui icon input">
    <input
      className="prompt"
      placeholder="Placeholder"
      type="text" />
    <i
      className="search icon" />
  </div>
</div>
`;
